src/scripts/main.js

```diff
@@ -1,3 +1,340 @@
+  });
+
+  for (let i = 0; i < newCellArr.length; i++) {
+    if (newCellArr[i] !== cellsArr[i]) {
+      createNumber();
+      break;
+    };
+  };
+
+  // Умова для перемоги
+  victory(newCellArr);
+
+  // Умова для програшу
+  loss(newCellArr);
+
+  addClass([...document.querySelectorAll('.field-cell')]);
+  scoreInGame.innerText = score;
+};
+
+// перемога
+function victory(valuesInCells) {
+  if (valuesInCells.find(cell => cell === '2048')) {
+    messegeStart.classList.add('hidden');
+    messegeLose.classList.add('hidden');
+    messegeWin.classList.remove('hidden');
+
+    document.removeEventListener('keydown', keydownListener);
+  };
+}
+
+// програш
+function loss(valuesInCells) {
+  const emptyCells = valuesInCells.filter(cell => cell === '');
+
+  if (emptyCells.length === 0) {
+    const horisontalNembers = {};
+    const verticalNumbers = {};
+
+    const allCells = document.querySelectorAll('td');
+
+    if (document.querySelector('tbody').classList.contains('tbody-columns')) {
+      for (let i = 0; i < [...allCells].length; i++) {
+        if (!horisontalNembers[allCells[i].cellIndex]) {
+          horisontalNembers[allCells[i].cellIndex] = [];
+        }
+        horisontalNembers[allCells[i].cellIndex].push(allCells[i].innerText);
+
+        if (!verticalNumbers[allCells[i].parentElement.rowIndex]) {
+          verticalNumbers[allCells[i].parentElement.rowIndex] = [];
+        };
+
+        verticalNumbers[allCells[i].parentElement.rowIndex]
+          .push(allCells[i].innerText);
+      };
+    } else {
+      for (let i = 0; i < [...allCells].length; i++) {
+        if (!verticalNumbers[allCells[i].cellIndex]) {
+          verticalNumbers[allCells[i].cellIndex] = [];
+        };
+        verticalNumbers[allCells[i].cellIndex].push(allCells[i].innerText);
+
+        if (!horisontalNembers[allCells[i].parentElement.rowIndex]) {
+          horisontalNembers[allCells[i].parentElement.rowIndex] = [];
+        };
+
+        horisontalNembers[allCells[i].parentElement.rowIndex]
+          .push(allCells[i].innerText);
+      };
+    }
+
+    let possibleNextStep = 0;
+
+    for (const row in horisontalNembers) {
+      for (let i = 1; i < horisontalNembers[row].length - 1; i++) {
+        if (horisontalNembers[row][i] === horisontalNembers[row][i - 1]
+          || horisontalNembers[row][i] === horisontalNembers[row][i + 1]) {
+          possibleNextStep++;
+        }
+      }
+    }
+
+    for (const row in verticalNumbers) {
+      for (let i = 1; i < verticalNumbers[row].length - 1; i++) {
+        if (verticalNumbers[row][i] === verticalNumbers[row][i - 1]
+          || verticalNumbers[row][i] === verticalNumbers[row][i + 1]) {
+          possibleNextStep++;
+        }
+      }
+    };
+
+    if (possibleNextStep === 0) {
+      messegeLose.classList.remove('hidden');
+      messegeStart.classList.add('hidden');
+      messegeWin.classList.add('hidden');
+      document.removeEventListener('keydown', keydownListener);
```

Comment:
Move such logic to a function with appropriate name. Overall, I recommend to move a lot of conditions inside of "if" statements, some logic that can be easily described as a function name etc., to separate functions. Then code becomes easily readable and is easy to support.
https://refactoring.guru/extract-method